### PR TITLE
add module example to std.container.array

### PR DESCRIPTION
**Pull request by wilzbach**
adds a top-level module example to `std.container.array` - ping @schveiguy

### CODE REVIEW

File: std/container/array.d [hunk 1]
```diff
@@ -27,6 +27,60 @@ import std.algorithm : move;
...
+    // insertion
+    arr.insertBefore(arr[1..$], 1);
+    assert(arr.front == 0);
+    assert(arr.length == 4);
+
+    arr.insertBack(4);
+    assert(arr.back == 4);
+    assert(arr.length == 5);
+
+    // set elemts
```
Comment (burner) [inline std/container/array.d]: elemts -> elements

File: std/container/array.d [hunk 2]
```diff
@@ -27,6 +27,60 @@ import std.algorithm : move;
...
+    arr[1] *= 42;
+    assert(arr[1] == 42);
+}
+
+///
+unittest
+{
+    import std.algorithm.comparison: equal;
```
Comment (burner) [inline std/container/array.d]: ": " -> " : "

**Comment by wilzbach** [comment]
@burner - comments addressed.

**Comment by burner** [comment]
Auto-merge toggled on

**Pull request merged**